tests/e2e/network/service_annotations.go

```diff
@@ -195,6 +196,80 @@ var _ = Describe("Service with annotation", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(ret).To(Equal(true), "external ip %s is not in the target subnet %s", ip, newSubnetCIDR)
 	})
+	It("can be bound to the mode of load balancer", func() {
+		mode_name := "validate_lb_mode"
+		annotation := map[string]string{
+			azure.ServiceAnnotationLoadBalancerMode: mode_name,
+		}
+
+		service := createLoadBalancerServiceManifest(cs, serviceName, annotation, labels, ns.Name, ports)
+		_, err := cs.CoreV1().Services(ns.Name).Create(service)
+		Expect(err).NotTo(HaveOccurred())
+		utils.Logf("Successfully created LoadBalancer service " + serviceName + " in namespace " + ns.Name)
+
+		if mode_name == "" {
+			utils.Logf("Service mode not set or set to default")
+		}else if mode_name == "auto" {
+			utils.Logf("Service mode set to auto")
+		}else{
+		utils.Logf("Service mode configured to %s",mode_name)
+		}
+		//get nodelist
+		v,err := utils.GetAgentNodes(cs)
+		//get providerID specific to an agentnode
+		Expect(err).NotTo(HaveOccurred())
+			if len(v) >= 2 {
+				vmss_name := make([]string, len(v))
```

Comment:
Use  `"k8s.io/apimachinery/pkg/util/sets"` for vmss names, e.g.

```go
vmssNames := sets.NewString()
...
vmssNames.Insert(newName)
```

Reply:
Is this a slice?

Reply:
this is a set